safe-global/web-core
# feat: tx filter

## What it solves

Adding the transaction filter.

## How this PR fixes it

This adds the filter form to the top of the historical transaction list. The URL is used as the single source of truth for which fields to query. When submitting the filter form, it appends search parameter to the URL which `useLoadTxHistory`/`useTxHistory` reacts to.

Note: there is one visual bug that renders the filter form partially down the page and doesn't unmount it sometimes that needs to be looked into.

## How to test it

Open the historical transaction list and observe the filter button, clicking which will open the transaction filter. Interacting with the form should work as expected, with validation and submitting it will log the filter to the console, as well as append it to the URL. Submitting the form will render filtered transactions.

## Screenshots
![filter interface](https://user-images.githubusercontent.com/20442784/186700739-e5b74ed5-5664-4518-bc8d-75e4a0080053.gif)
